### ISSUE TITLE
Don't unwrap when creating values while parsing edn

### ISSUE THREAD

**Issue by victorporof** [mozilla/mentat#230]
E.g. https://github.com/mozilla/mentat/blob/ab041291fb1d806fc1e9c09ecac4a931289bd38f/edn/src/edn.rustpeg#L45 and in a few other places.

Panicking the entire thread while parsing is probably a bad idea? This would be easily recoverable and a good place to add syntax check info.

**Comment by rnewman** [comment]
I agree, but this is as good a place as any to note three observations:

- Many of the `unwrap`s in that parser are totally appropriate. That is, if we read an optional `sign`, then any number of `digit`s, followed by an `N`, and it *doesn't* parse to a `BigInt`, we are in deep shit, and we should crash hard. It means either something is in a bad state, or we have a serious coding error. `panic` is an appropriate way to deal with the world being wrong.
- In an embedded layer like this, it can be more appropriate to crash than to handle errors. If we crash, we'll get a crash report, rather than having some part of Firefox silently stop working and never receiving a good bug report.
- It's not unreasonable to take the Erlang approach: parse queries on the same thread that's going to run them, and if it fails to parse or otherwise dies, well, that query didn't run to completion. This is the approach we will need to take for query timeouts, because SQLite doesn't support interrupting a query — we will have a monitor simply kill the thread.

Death, alas, is a part of life.
